### ISSUE TITLE
Feature Request - Support Spine 4 (with WebP image export)

### ISSUE THREAD

**Issue by Ronsku** [cocos/cocos-engine#10194]
Finally WebP has been added as support for Spine in 4.1.20-beta version. When this verison comes out of beta it would be amazing for Cocos to finally also upgrade to Support Spine 4. It would be good at the same time to take a look at the Typings for Spine, because they are not so up to date, even in the latest CC 4.3.1.

![image](https://user-images.githubusercontent.com/7983358/155812603-9ab7ca0f-47ad-499d-bb20-8ad64b6c3549.png)

Thanks!

**Comment by jareguo** [comment]
Thanks. Consider upgrading in 3.5 @holycanvas

**Comment by SantyWang** [comment]
@zhakesi Please research about Spine 4.x

**Comment by SantyWang** [comment]
After our research, we found that Spine 4 is not compatible with the resources generated by the Spine 3.x editor, which means that once upgraded, most users' projects may not work properly, so this will need to wait until the Creator's multi-version plugin mechanism is perfected, and then developers will be able to choose the version of Spine they want to use. maybe in v3.7

**Comment by Ronsku** [comment]
> After our research, we found that Spine 4 is not compatible with the resources generated by the Spine 3.x editor, which means that once upgraded, most users' projects may not work properly, so this will need to wait until the Creator's multi-version plugin mechanism is perfected, and then developers will be able to choose the version of Spine they want to use. maybe in v3.7

I see, that makes sense. Would be perfect to be able to chose which version of spine you would run with from the Feature Cropping like this
![image](https://user-images.githubusercontent.com/7983358/167853753-9d279710-d7e1-4a84-a3c3-a524f73d491f.png)

Just need to make sure it doesn't put everything into the CC.js package like it currently looks like it is doing, since engine is massive and slows down the loading: https://github.com/cocos/cocos-engine/issues/10354

**Comment by SantyWang** [comment]
yeah, it's a good advise, but who knows? perhaps creator will have own package manager 😃

**Comment by Ronsku** [comment]
Any plans on Spine Version 4 support?

**Comment by Canvasfull** [comment]
In the future, we will support plug-in management spine, which is still under planning and development. The fastest version is coming. cocos creator 3.9+. 

Like the picture below:
<img width="341" alt="Screenshot 2023-08-31 at 18 26 21" src="https://github.com/cocos/cocos-engine/assets/2398658/52421b34-5875-4b1a-b933-2f6d52f34d2b">

**Comment by Ronsku** [comment]
> In the future, we will support plug-in management spine, which is still under planning and development. The fastest version is coming. cocos creator 3.9+.

Awesome, looks really promising! 🤩 Good job! 💪 
Looking forward to this and for Spine V4 support 😊

**Comment by Ronsku** [comment]
> In the future, we will support plug-in management spine, which is still under planning and development. The fastest version is coming. cocos creator 3.9+.

We're closing up on 3.9, still planned for this?

**Comment by bofeng-song** [comment]
Currently, it is still in the planning stage.

**Comment by bofeng-song** [comment]
> > In the future, we will support plug-in management spine, which is still under planning and development. The fastest version is coming. cocos creator 3.9+.
> 
> We're closing up on 3.9, still planned for this?

Currently, it is still in the planning stage.

**Comment by antn9x** [comment]
I want to vote for this feature release soon.

**Comment by lagatun54** [comment]
I want to vote for this feature release soon.

**Comment by AdminTinyTech** [comment]
Any updates on Spine v4 support? Request any latest developments on this.

**Comment by Lykdraft** [comment]
Yeah. Spine is now 4.2. We're moving way too slow here...

**Comment by bofeng-song** [comment]
The version of Spine 4.2 is currently under development. This is the [PR](https://github.com/cocos/cocos-engine/pull/18325)

**Comment by Lykdraft** [comment]
I see Spine V4.2 support is merged. Good work!

**Comment by bofeng-song** [comment]
Some newly added interfaces in Spine 4.2 may not be exported and will be improved in subsequent versions. If you encounter any issues, please provide feedback here. Thank you.

**Comment by devtinytech1** [comment]
This is so great to have spine 4.2 support being available with Cocos. Thanks for working on this @bofeng-song !! Any idea when 3.8.6 will be available to download?
